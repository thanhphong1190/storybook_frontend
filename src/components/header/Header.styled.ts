import styled from 'styled-components'
import tw from 'twin.macro'
import { Button } from '../base'

export const Container = styled.div`
  ${tw`container mx-auto flex flex-wrap justify-center xl:justify-between items-center p-2 flex-col lg:flex-row`}
`

export const CTAGroup = styled.div`
  ${tw`order-1 xl:order-none flex flex-wrap items-center justify-center text-sm`}
`
export const ActionTitle = styled.span`
  ${tw`uppercase text-white font-bold mr-1 md:mr-3 text-grey__medium xl:text-white text-xs lg:text-base`}
`
export const SearchGroup = styled.div`
  ${tw`flex flex-row items-stretch w-full md:w-auto`}
  button {
    ${tw`ml-3 flex`}
  }
  input {
    min-width: 0;
    ${tw`bg-blue__grey flex-1 px-1 lg:px-2 lg:w-1/3 xl:w-64 xl:text-white xl:placeholder-white`}
  }
`

export const HeaderTitle = styled.div`
  ${tw`order-none md:order-none flex flex-row items-center`}
`

export const TitleText = styled.span`
  ${tw`font-bold text-sm lg:text-2xl uppercase`}
  ${({ color }) => (color == 'orange' ? tw`text-orange` : tw`text-white`)}
  & {
    &:last-child {
      ${tw`ml-1`}
    }
  }
`

export const HeaderSection = styled.section`
  ${tw`w-full bg-blue__grey`}
`

export const SearchButton = styled(Button)`
  ${tw`py-2`}
`

export default HeaderSection
