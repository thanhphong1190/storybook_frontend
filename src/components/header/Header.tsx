import { TextInput } from '../../styles/global.styles'
import React from 'react'
import * as S from './Header.styled'

export interface HeaderProps {
  CTATitle: string
  Titles: [string, string]
  buttonLabel: string
  buttonAction: string
  searchInputProps: object
}

export const Header = (props: HeaderProps) => {
  const {
    CTATitle,
    Titles: [orangeText, whiteText],
    buttonLabel,
    searchInputProps
  } = props
  return (
    <S.HeaderSection>
      <S.Container>
        <S.CTAGroup>
          <S.ActionTitle>{CTATitle}</S.ActionTitle>
          <S.SearchGroup>
            <TextInput {...searchInputProps} />
            <S.SearchButton color='blue'>{buttonLabel}</S.SearchButton>
          </S.SearchGroup>
        </S.CTAGroup>
        <S.HeaderTitle>
          <S.TitleText color='orange'>{orangeText}</S.TitleText>
          <S.TitleText>{whiteText}</S.TitleText>
        </S.HeaderTitle>
      </S.Container>
    </S.HeaderSection>
  )
}
export default Header
