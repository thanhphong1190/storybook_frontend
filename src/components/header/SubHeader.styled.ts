import styled from 'styled-components'
import tw from 'twin.macro'
import Select from 'react-select'

export const SubHeaderWrapper = styled.div`
  ${tw`relative flex flex-col-reverse items-center md:flex-row w-full font-display`}
`

export const DropdownStyled = styled(Select)`
  ${tw`uppercase italic`}
  min-width: 10rem;
  & > div {
    ${tw`rounded-none border-blue`}
  }
`

export const SubHeaderStyled = styled.h2<{ separator?: boolean, hasDropdown?: boolean }>`
  ${tw`text-blue__dark flex-1 text-center text-2xl py-4 md:pl-40 uppercase font-extrabold`}

  ${props => !props.hasDropdown && tw`md:pl-0`}

  &::after {
    ${props => props.separator && tw`absolute bottom-0 h-1 bg-orange`}
    content: " ";
    left: 50%;
    width: 185px;
    transform: translateX(-50%) rotate(-4deg);
  }
`
