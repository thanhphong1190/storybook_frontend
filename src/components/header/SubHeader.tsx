import { SelectOption } from '../../services/api/types'
import { SORT_OPTIONS, SORT_PLACEHODLER } from '../../services/api/constants'
import {
  SubHeaderWrapper,
  SubHeaderStyled,
  DropdownStyled
} from './SubHeader.styled'

export interface SubHeaderProps {
  subHeaderText: string
  showDropdown: boolean
  showSeparator: boolean
  dropdownOptions?: Array<SelectOption>
  dropdownPlaceholder?: string
}

export const SubHeader = (props: SubHeaderProps) => {
  const {
    subHeaderText,
    dropdownPlaceholder,
    dropdownOptions,
    showDropdown,
    showSeparator
  } = props

  return (
    <SubHeaderWrapper>
      <SubHeaderStyled hasDropdown={showDropdown} separator={showSeparator}>
        {subHeaderText}
      </SubHeaderStyled>

      {showDropdown ? (
        <DropdownStyled
          options={dropdownOptions || SORT_OPTIONS}
          placeholder={dropdownPlaceholder || SORT_PLACEHODLER}
        />
      ) : (
        ''
      )}
    </SubHeaderWrapper>
  )
}

export default SubHeader
