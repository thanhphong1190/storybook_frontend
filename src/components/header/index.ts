export { Header, HeaderProps } from './Header'
export { SubHeader, SubHeaderProps } from './SubHeader'
