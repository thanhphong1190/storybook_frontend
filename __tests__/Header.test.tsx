import { Header, HeaderProps } from '../src/components/header/Header'
import renderer from 'react-test-renderer'

const props: HeaderProps = {
  CTATitle: 'Find your plus fitness location',
  Titles: ['24/7 open gym - ', 'no lock in contracts'],
  buttonLabel: 'Search',
  buttonAction: '#',
  searchInputProps: {
    placeholder: 'Type your suburb or post code',
  },
}

it('renders correctly', () => {
  const tree = renderer.create(<Header {...props} />).toJSON()
  expect(tree).toMatchSnapshot()
})
