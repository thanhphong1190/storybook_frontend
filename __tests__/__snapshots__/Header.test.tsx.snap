// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  className="Headerstyled__HeaderSection-ihh9tu-6 fHvlyo"
>
  <div
    className="Headerstyled__Container-ihh9tu-0 jUjIsR"
  >
    <div
      className="Headerstyled__CTAGroup-ihh9tu-1 eiCaXq"
    >
      <span
        className="Headerstyled__ActionTitle-ihh9tu-2 erABvG"
      >
        Find your plus fitness location
      </span>
      <div
        className="Headerstyled__SearchGroup-ihh9tu-3 jKxLKy"
      >
        <input
          className="globalstyles__TextInput-sc-1nc69wi-7 jfclbV"
          placeholder="Type your suburb or post code"
        />
        <button
          className="Button-lxhj0a-0 Headerstyled__SearchButton-ihh9tu-7 dxrwLE"
          color="blue"
        >
          Search
        </button>
      </div>
    </div>
    <div
      className="Headerstyled__HeaderTitle-ihh9tu-4 cnaWgv"
    >
      <span
        className="Headerstyled__TitleText-ihh9tu-5 juWinu"
        color="orange"
      >
        24/7 open gym - 
      </span>
      <span
        className="Headerstyled__TitleText-ihh9tu-5 gerIAq"
      >
        no lock in contracts
      </span>
    </div>
  </div>
</section>
`;
